### PR TITLE
Specify the resource type when uploading one.

### PR DESCRIPTION
**Pull request by facundobatista**
This is because the API changed, specifying the type for files is optional (for now), but we will need it soon to indicate oci-image types.

Also fixed/improved two help-related details in other commands.

### CODE REVIEW

**Review by cjp256 — APPROVED** [review]
LGTM

File: charmcraft/commands/store/store.py [hunk 1]
```diff
@@ -184,10 +187,10 @@ def upload(self, name, filepath):
         endpoint = '/v1/charm/{}/revisions'.format(name)
         return self._upload(endpoint, filepath)
 
-    def upload_resource(self, charm_name, resource_name, filepath):
+    def upload_resource(self, charm_name, resource_name, resource_type, filepath):
         """Upload the content of filepath to the indicated resource."""
         endpoint = '/v1/charm/{}/resources/{}/revisions'.format(charm_name, resource_name)
-        return self._upload(endpoint, filepath)
+        return self._upload(endpoint, filepath, extra_fields={'type': resource_type})
```
Comment (sergiusens) [inline charmcraft/commands/store/store.py:193]: what happened to reading the value for metadata?

**Review by sergiusens — APPROVED** [review]
Looks good. Just a question to keep track of changes

File: charmcraft/commands/store/store.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/store.py:193]: Don't know, will find out,

**Pull request merged**